sql-parser/src/main/java/io/crate/sql/Literals.java

```diff
@@ -24,14 +24,140 @@
 
 import com.google.common.annotations.VisibleForTesting;
 
+import java.util.function.Predicate;
+
 public class Literals {
 
+    private static final String EPSILON = "E";
+    private static final String ESCAPED_UNICODE_ERROR = "Invalid Unicode escape (must be \\uXXXX or \\UXXXXXXXX)";
+
     public static String quoteStringLiteral(String literal) {
         return "'" + escapeStringLiteral(literal) + "'";
     }
 
+    public static String escapeAndQuoteStringLiteral(String literal) {
+        return EPSILON + Literals.quoteStringLiteral(literal);
+    }
+
     @VisibleForTesting
     static String escapeStringLiteral(String literal) {
         return literal.replace("'", "''");
     }
+
+    public static String replaceEscapedChars(String input) {
+        int length = input.length();
+        if (input.length() <= 1) {
+            return input;
+        }
+
+        StringBuilder builder = new StringBuilder(length);
+        int endIdx;
```

Comment:
Could probably reduce the scope of `endIdx` to the places where it is used, to make it clear that it isn't used outside of the specific case blocks.

Reply:
I would rather not change this: `endIdx` is used in 3 main use cases. 
In addition, this would require to either use different name in each case block, or use anonymous blocks or declare it only in the first case where it's used (which I personally find more confusing).
In addition, it's in a for loop, so I'd rather declare it once.